src/BudgetManager/BudgetManager.DataAccess/MongoDbAccess/Repositories/TransactionRepository.cs

```diff
@@ -1,16 +1,33 @@
-﻿using BudgetManager.Model;
+﻿using BudgetManager.DataAccess.MongoDbAccess.Interfaces;
+using BudgetManager.Model;
+using BudgetManager.Model.Enums;
 using BudgetManager.Shared.DataAccess.MongoDB.BaseImplementation;
 using BudgetManager.Shared.DataAccess.MongoDB.DatabaseSettings;
-using MongoDB.Bson;
 using MongoDB.Driver;
 
 namespace BudgetManager.DataAccess.MongoDbAccess.Repositories
 {
-    public class TransactionRepository : BaseRepository<Transaction>
+    public class TransactionRepository : BaseRepository<Transaction>, ITransactionRepository
     {
         public TransactionRepository(IMongoDbSettings settings, IMongoClient client)
             : base(settings, client)
         {
         }
+
+        public async Task<IEnumerable<Transaction>> GetListByOperationAsync(Guid userId, OperationType operationType, CancellationToken cancellationToken)
+        {
+            var filterUser = Builders<Transaction>.Filter.Eq(x => x.UserId, userId);
+            var filterOperation = Builders<Transaction>.Filter.Eq(x => x.TransactionType, operationType);
+            var filter = Builders<Transaction>.Filter.And(filterUser, filterOperation);
+            var result = await _collection.Find(filter).SortBy(x => x.DateOfTransaction).ToListAsync(cancellationToken);
+            return result;
+        }
+
+        public async Task<IEnumerable<Transaction>> GetListByUserIdAsync(Guid userId, CancellationToken cancellationToken)
+        {
+            var filter = Builders<Transaction>.Filter.Eq(x => x.UserId, userId);
+            var result = await _collection.Find(filter).SortBy(x => x.DateOfTransaction).ToListAsync(cancellationToken);
+            return result;
```

Comment:
Why are you create creating methods that's alreaty exist in base repo?

You need just to throw filters in filtration method of base repo
And use handlers for this aproach

Reply:
Because FilterBy method in Base Repo retuens IEnumarable<T>. SortBy is native method of Mongo.Driver

Reply:
Currently now there are two main collection works with BaseRepo: User and Transaction. other entities are being sorted and should be sorted in handler. because they do not have their own collection

Reply:
Ok)
In that case let's do in this way)
Good job